### ISSUE TITLE
GraphQL: Unable to parse variable value 0x... as an Bytes32 

### ISSUE THREAD

**Issue by bl0up** [hyperledger/besu#4441]
### Description
I have a besu node running and I am trying to query transactions using their hash: 

```
query getTransaction($hash: Bytes32!) {
  transaction(hash: $hash) {
    status
    nonce
    hash
  }
}
```
I define the hash variable as follows:
```
{ "hash": "0x3ce2249f435a4f97b25141a147076206d8922d5c9b8ded9f4cbfa8b427f4fd16"}
```

When I run the query, I get the following error: 
```
{
  "errors": [
    {
      "message": "invalid value : invalid value : Unable to parse variable value 0x3ce2249f435a4f97b25141a147076206d8922d5c9b8ded9f4cbfa8b427f4fd16 as an Bytes32"
    }
  ]
}
```

Simply writing 
```
transaction(hash: "0x3ce2249f435a4f97b25141a147076206d8922d5c9b8ded9f4cbfa8b427f4fd16"){
  status
  nonce
  hash
}
```
works fine. 

### Steps to Reproduce (Bug)
1. Setup a private besu node
2. deploy a smart contract and send transaction
3. Try to use the `getTransaction` query

**Expected behavior:**
It should return the status, nonce and hash of the transaction with the corresponding hash
**Actual behavior:**]
It returns an error message saying "unable to parse variable value"

**Frequency:**
100%

**Comment by macfarla** [comment]
Hi, thanks for your interest in besu. Since the query works fine in the raw form I suspect it's how you're assigning the value to the `hash` variable. How are you setting up the variable and then passing the query to Besu?

**Comment by macfarla** [comment]
Does this give any clues https://stackoverflow.com/questions/46491123/string-parameter-not-automatically-parsing-into-bytes32-when-used-with-form

**Comment by shemnon** [comment]
There are some bugs in scalar conversions.  PR #4522 should fix it (on the 22.10.x train).
